aiohttp/client.py

```diff
@@ -167,6 +177,7 @@ class ClientTimeout:
 
 # https://www.rfc-editor.org/rfc/rfc9110#section-9.2.2
 IDEMPOTENT_METHODS = frozenset({"GET", "HEAD", "OPTIONS", "TRACE", "PUT", "DELETE"})
+HTTP_SCHEMA_SET = frozenset({"http", "https", ""})
```

Comment:
@bdraco @setla would augmenting this make it work?

Reply:
I will analyze how its being used in the case I posted above when I get back home next week and provide additional detail

Reply:
As a possible future improvement, I'd rename the constant to something like `SUPPORTED_URI_SCHEMAS`. Especially, if it'll stop doing listing http only.

Reply:
@webknjaz  @bdraco  Mentioned exception is related because it's raised here https://github.com/aio-libs/aiohttp/pull/6722/files#diff-f334e752b4894ef951105572ab8b195aeb8db90eb6e48b1dfbd9a01da4c854f5R420 but it's also used in redirects check https://github.com/aio-libs/aiohttp/pull/6722/files#diff-f334e752b4894ef951105572ab8b195aeb8db90eb6e48b1dfbd9a01da4c854f5R647. If we don't want regression, probably first mentioned place should be deleted

Reply:
We could have two constants where one lists ws/wss and the check asserts against both in the first place but not the second one. Are websockets supposed to be disallowed in redirects?

Reply:
Which means there should probably be an `allow_redirects=False` in here:
https://github.com/aio-libs/aiohttp/blob/dc38630b168a169139974617d75e176530c91696/aiohttp/client.py#L848-L860

Reply:
@bdraco did you end up deciding how this should be fixed?

Reply:
I haven't had a free moment to get back to this. After the Home Assistant beta tomorrow and initial feedback I should be able to come back to it

Reply:
In this case the call that is failing is:

`await session.ws_connect('wss://.....')` 

Docs say
```
url – Websocket server url, [URL](https://yarl.aio-libs.org/en/stable/api/#yarl.URL) or [str](https://docs.python.org/3/library/stdtypes.html#str) that will be encoded with [URL](https://yarl.aio-libs.org/en/stable/api/#yarl.URL) (see [URL](https://yarl.aio-libs.org/en/stable/api/#yarl.URL) to skip encoding).
```

So we definitely have a regression.

looking how to fix it now

Reply:
Since  `ws_connect` always calls into `self.request` think the simplest solution to avoid a breaking change is:

```diff
diff --git a/aiohttp/client.py b/aiohttp/client.py
index 8d8d13f2..d608c009 100644
--- a/aiohttp/client.py
+++ b/aiohttp/client.py
@@ -178,7 +178,7 @@ DEFAULT_TIMEOUT: Final[ClientTimeout] = ClientTimeout(total=5 * 60)
 
 # https://www.rfc-editor.org/rfc/rfc9110#section-9.2.2
 IDEMPOTENT_METHODS = frozenset({"GET", "HEAD", "OPTIONS", "TRACE", "PUT", "DELETE"})
-HTTP_SCHEMA_SET = frozenset({"http", "https", ""})
+HTTP_SCHEMA_SET = frozenset({"http", "https", "ws", "wss", ""})
 
 _RetType = TypeVar("_RetType")
 _CharsetResolver = Callable[[ClientResponse, bytes], str]
```

Reply:
I think we probably want that in the _request() check, but probably still want to limit it to http/https in the redirects, and disable redirects in _ws_connect() (as mentioned above).

Reply:
That makes sense to me.  So it seems we need to different constants

Reply:
this should include wss:// as well as this breaks discord.py entirely.

Reply:
I probably lost track of this, maybe nobody got round to making the change discussed above.